adafruit_led_animation/color.py

```diff
@@ -64,3 +64,38 @@ def colorwheel(pos):
             return 0, int(255 - pos * 3), int(pos * 3)
         pos -= 170
         return int(pos * 3), 0, int(255 - (pos * 3))
+
+
+def calculate_intensity(color, intensity=1.0):
+    """
+    Takes a RGB[W] color tuple and adjusts the intensity.
+    :param float intensity:
+    :param color: color value (tuple, list or int)
+    :return: color
+    """
+    if isinstance(color, int):
+        return (
+            (int((color & 0xFF0000) * intensity) & 0xFF0000)
+            | (int((color & 0xFF00) * intensity) & 0xFF00)
+            | (int((color & 0xFF) * intensity) & 0xFF)
+        )
+
+    if len(color) == 3:
+        return (
+            int(color[0] * intensity),
+            int(color[1] * intensity),
+            int(color[2] * intensity),
+        )
+    elif len(color) == 4 and isinstance(color[3], float):
+        return (
+            int(color[0] * intensity),
+            int(color[1] * intensity),
+            int(color[2] * intensity),
+            color[3],
+        )
+    return (
+        int(color[0] * intensity),
+        int(color[1] * intensity),
+        int(color[2] * intensity),
+        int(color[3] * intensity),
+    )
```

Comment:
You could check for the 3/4 case specially and otherwise do:
```python
    return(int(p * intensity) for p in color)
```
for the regular 3 and 4 cases.

Reply:
Hi @dhalbert 
The reason I used the long form was to speed things up.  In 10,000 iterations on a Feather NRF.  I looked for a faster way to reconvert back to int, but didn't see one.  I've been wondering if there is an integer-only way to multiply by 0-1.0.

- the 3-way conditional took 3.2s
- the tuple comprehension took 5.6s
- using a list comprehension took 4.7s

Reply:
You could scale things to do integer arithmetic, but I don't think floating point vs integer arithmetic is significant on M4 processors. I'll approve this, based on your thorough timing.